src/main/scala/scalaz/analytics/AnalyticsModule.scala

```diff
@@ -78,16 +97,16 @@ trait AnalyticsModule {
   }
 
   /**
-   * The Operations supported by the core DataStream abstraction in scalaz-analytics.
+   * The dataset/datastream operations of scalaz-analytics
    */
-  trait SetOperations {
-    def union[A](l: DataStream[A], r: DataStream[A]): DataStream[A]
-    def intersect[A](l: DataStream[A], r: DataStream[A]): DataStream[A]
-    def except[A](l: DataStream[A], r: DataStream[A]): DataStream[A]
-    def distinct[A](d: DataStream[A]): DataStream[A]
-    def map[A, B](d: DataStream[A])(f: A =>: B): DataStream[B]
-    def sort[A](d: DataStream[A]): DataStream[A]
-    def distinctBy[A, B](d: DataStream[A])(by: A =>: B): DataStream[A]
+  trait Ops[F[_]] {
+    // Unbounded
+    def map[A, B](ds: F[A])(f: A =>: B): F[B]
+    def filter[A, B](ds: F[A])(f: A =>: Boolean): F[A]
+
+    // Bounded
+    def fold[A, B](ds: F[A])(window: Window)(initial: A =>: B)(f: (B, A) =>: B): F[B]
```

Comment:
Not 100% sure about this signature. Is the `initial: A =>: B` correct?

Reply:
Additionally - to make this usable for user code, we're going to need to flesh out the tuple syntax